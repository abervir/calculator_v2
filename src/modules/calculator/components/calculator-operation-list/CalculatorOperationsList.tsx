import React from 'react';

import ReactDropdown from 'react-dropdown';
import { CalculatorOperations } from '@modules/calculator/constants';
import { MathOperation } from '@modules/calculator/enums';
import { CalculatorOperationsListItem } from '../calculator-operation-list-item/CalculatorOperationsListItem';

interface Props {
  onChangeOperation: (value: MathOperation) => void;
}

export const CalculatorOperationsList = (props: Props) => {
  const { onChangeOperation } = props;

  const dropdownOptions = () => {
    return CalculatorOperations.map(({ id, title, icon }) => ({
      value: id.toString(),
      label: <CalculatorOperationsListItem title={title} icon={icon} />,
    }));
  };

  return (
    <div className='calculator-operations-list'>
      <ReactDropdown
        onChange={({ value }) => onChangeOperation(value as MathOperation)}
        options={dropdownOptions()}
        placeholder='Select an option'
      />
    </div>
  );
};
