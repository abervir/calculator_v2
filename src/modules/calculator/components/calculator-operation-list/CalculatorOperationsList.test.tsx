import { shallow } from 'enzyme';
import React from 'react';

import { CalculatorOperationsList } from './CalculatorOperationsList';

describe('CalculatorOperationsList', () => {
  it('should render CalculatorOperationsList component', () => {
    expect(
      shallow(<CalculatorOperationsList onChangeOperation={() => {}} />)
    ).toMatchSnapshot();
  });
});
