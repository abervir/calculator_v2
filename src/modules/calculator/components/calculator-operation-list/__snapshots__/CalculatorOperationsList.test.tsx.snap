// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CalculatorOperationsList should render CalculatorOperationsList component 1`] = `
<div
  className="calculator-operations-list"
>
  <Dropdown
    baseClassName="Dropdown"
    onChange={[Function]}
    options={
      Array [
        Object {
          "label": <CalculatorOperationsListItem
            icon="icon-plus-sign"
            title="Sum"
          />,
          "value": "PLUS",
        },
        Object {
          "label": <CalculatorOperationsListItem
            icon="icon-divide-sign"
            title="Divide"
          />,
          "value": "DIVIDE",
        },
        Object {
          "label": <CalculatorOperationsListItem
            icon="icon-remainder-of-division-sign"
            title="Remainder of a division"
          />,
          "value": "REMAINDER_OF_DIVISION",
        },
        Object {
          "label": <CalculatorOperationsListItem
            icon="icon-prime-number-sign"
            title="Max prime number between two operands"
          />,
          "value": "MAX_PRIME_NUMBER",
        },
      ]
    }
    placeholder="Select an option"
  />
</div>
`;
